Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
     {
-        osDelayUntil(&prevWakeTime, SLOW_LOG_DATA_PERIOD);
-
-        if (getCurrentFlightPhase() != entryPhase)
-        {
-            // New phase has started, exit low frequency logging
-            return;
-        }
-
-        buildLogEntry(data, buffer);
-        writeAllDataToEEPROM(data, memAddress, timeout);
-
+        theLogEntry.gps_time = data->gpsData_->time_;
+        theLogEntry.latitude_degrees = data->gpsData_->latitude_.degrees_;
+        theLogEntry.latitude_minutes = data->gpsData_->latitude_.minutes_;
+        theLogEntry.longitude_degrees = data->gpsData_->longitude_.degrees_;
+        theLogEntry.longitude_minutes = data->gpsData_->longitude_.minutes_;
+        theLogEntry.antennaAltitude = data->gpsData_->antennaAltitude_.altitude_;
+        theLogEntry.geoidAltitude = data->gpsData_->geoidAltitude_.altitude_;
+        theLogEntry.altitude = data->gpsData_->totalAltitude_.altitude_;
+        osMutexRelease(data->gpsData_->mutex_);
     }
+    theLogEntry.currentFlightPhase = getCurrentFlightPhase();
+    theLogEntry.tick = HAL_GetTick();
 }
 
-void highFrequencyLogToSdRoutine(AllData* data, char* buffer)
+/**
+ * @brief Used for logging entries to the EEPROM. First updates the struct, then writes it
+ * the wait times are all managed in the main for loop so highFrequencyLog = lowFrequencyLog
+ * @param data, pointer to AllData Struct sent to updateLogEntry to update the logEntry struct
+ */
+void logEntryOnceRoutine(AllData* data)
 {
-    // Get card mounted
-    uint8_t mounted = 0;
-
-    while (!mounted)
-    {
-        // Keep trying, really need to log during this time
-        mounted = (f_mount(&fatfs, "SD:", 1) == FR_OK);
-
-        FlightPhase flightPhase = getCurrentFlightPhase();
-
-        if ( flightPhase != BURN &&
-                flightPhase != COAST &&
-                flightPhase != DROGUE_DESCENT)
-        {
-            // couldn't mount during important phases, too bad :(
-            return;
-        }
-    }
-
-    // Card mounted, start writing at high frequency
-    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 1);
-    uint32_t prevWakeTime = osKernelSysTick();
-
-    for (;;)
-    {   // Grab
-        osDelayUntil(&prevWakeTime, FAST_LOG_DATA_PERIOD);
-
-        FlightPhase flightPhase = getCurrentFlightPhase();
-
-        if ( flightPhase != BURN &&
-                flightPhase != COAST &&
-                flightPhase != DROGUE_DESCENT)
-        {
-            // done important phases, unmount card and exit high frequency logging
-            break;
-        }
-
-        buildLogEntry(data, buffer);
-
-        if (f_open(&file, fileName, FA_OPEN_APPEND | FA_READ | FA_WRITE) == FR_OK)
-        {
-            f_puts(buffer, &file);
-            f_close(&file); // close to save the file
-        }
-    }
-
-    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, 0);
-    f_mount(NULL, "SD:", 1);
+    updateLogEntry(data);
+    writeLogEntryToEEPROM(currMemAddress);
```

Comment:
Need to increment offset:
```
writeLogEntryToEEPROM(EEPROM_START_ADDRESS + logAddressOffset);
logAddressOffset += LOG_SIZE_BYTES;
```

Need to include space-saving logic here. Probably needs to be in a different change.